datacube/index/postgres/tables/_core.py

```diff
@@ -113,6 +113,19 @@ def _pg_exists(conn, name):
     return conn.execute("SELECT to_regclass(%s)", name).scalar() is not None
 
 
+def _pg_column_exists(conn, table, column):
+    """
+    Does a postgres object exist?
+    :rtype bool
+    """
+    return conn.execute("""
+                        select TRUE from pg_attribute
+                        where attrelid = to_regclass(%s)
+                        and attname = %s
+                        and not attisdropped
+                        """, table, column) is not None
```

Comment:
I think this always returns true (?). I tried a test update and it didn't make changes.

You could use a similar `select 1 .. scalar()` as is done below.